docs/DeveloperGuide.md

```diff
@@ -373,96 +373,249 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 (For all use cases below, the **System** is the `ContaX` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: List persons**
+#### Person-Related Use Cases
+![Person Use Case](images/UseCaseDiagramPersons.png)
 
-**MSS**
+**UC1: List Persons**
 
-1. User requests to list persons
-2. ContaX shows a list of persons
+**MSS**
 
+1. User requests to list persons.
+2. ContaX shows a list of persons.
     Use case ends.
 
-**Use case: Add a person**
+**UC2: Add Person**
 
 **MSS**
 
-1. User requests to add a person
-2. ContaX adds the person
-
+1. User requests to add a person.
+2. User enters details of the new person.
+3. ContaX adds the new person.
+4. ContaX shows that the person has been added successfully.
     Use case ends.
 
 **Extensions**
 
-* 1a. The command given is in an invalid format
+* 2a. ContaX detects that a required person attribute was not supplied.
+    * 2a1. ContaX shows an error message indicating that there is a missing required attribute.
+    * Use case ends.<br>&nbsp;
 
-    * 1a1. ContaX shows an error message indicating that the given command is of invalid format.
+* 2b. ContaX detects that a supplied attribute has an invalid value.
+    * 2b1. ContaX shows an error message indicating that the supplied parameter is invalid.
+    * 2b2. ContaX shows the expected allowed values.
+    * Use case ends.<br>&nbsp;
 
-    * Use case ends.
+* 2c. ContaX detects that the person already exists.
+    * 2c1. ContaX shows an error message indicating that the person already exists.
+    * Use case ends.<br>&nbsp;
 
-* 1b. A parameter given is in an invalid format
+**UC3: Delete Person**
 
-    * 1b1. ContaX shows an error message indicating that the given parameter is of invalid format.
+**MSS**
 
-    * Use case ends.
+1. User requests to **list persons (UC1)**.
+2. ContaX displays a list of persons.
+3. User requests to delete a person.
+4. User enters details to delete person.
+5. ContaX deletes the person.
+6. ContaX displays a message indicating that the person was successfully deleted.
+    Use case ends.
 
-* 1c. The person already exists
+**Extensions**
 
-    * 1c1. ContaX shows an error message indicating that the person already exists.
+* 1a. User filters person by attribute.
+  * 1a1. User **finds person (UC5)**.
+  * Use case resumes from step 2.<br>&nbsp;
 
-    * Use case ends.
+* 1b. User filters persons by tag keyword.
+  * 1b1. User **finds person by tag (UC10)**.
+  * Use case resumes from step 2.<br>&nbsp;
 
-**Use case: Delete a person**
+* 2a. ContaX has no persons to list.
+  * Use case ends.<br>&nbsp;
 
-**MSS**
+* 4a. ContaX detects that a required person attribute was not supplied.
+    * 4a1. ContaX shows an error message indicating that there is a missing required attribute.
+    * Use case ends.<br>&nbsp;
```

Comment:
Delete does not require any attributes, just the index

Reply:
Fixed in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/263/commits/b635bf79c7071c3d4393bb9cd86306ce5e7bfd5f